pkg/sources/github/github.go

```diff
@@ -273,6 +273,58 @@ func (s *Source) Init(aCtx context.Context, name string, jobID, sourceID int64,
 	return nil
 }
 
+// Validate is used by enterprise CLI to validate the Github config file.
+func (s *Source) Validate(ctx context.Context) []error {
+	var (
+		errs     []error
+		ghClient *github.Client
+		err      error
+	)
+	apiEndpoint := s.conn.Endpoint
+
+	switch cred := s.conn.GetCredential().(type) {
+	case *sourcespb.GitHub_BasicAuth:
+		s.httpClient.Transport = &github.BasicAuthTransport{
+			Username: cred.BasicAuth.Username,
+			Password: cred.BasicAuth.Password,
+		}
+		ghClient, err = createGitHubClient(s.httpClient, apiEndpoint)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	case *sourcespb.GitHub_Unauthenticated:
+		ghClient, err = createGitHubClient(s.httpClient, apiEndpoint)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	case *sourcespb.GitHub_Token:
+		s.githubToken = cred.Token
+
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: s.githubToken},
+		)
+		s.httpClient.Transport = &oauth2.Transport{
+			Base:   s.httpClient.Transport,
+			Source: oauth2.ReuseTokenSource(nil, ts),
+		}
+
+		ghClient, err = createGitHubClient(s.httpClient, apiEndpoint)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	default:
+		errs = append(errs, errors.Errorf("Invalid configuration given for source. Name: %s, Type: %s", s.name, s.Type()))
+	}
+
+	// Run a simple query to check if the client is valid
+	_, _, err = ghClient.Users.Get(ctx, "")
```

Comment:
ah, ok. do you want to try using the client if there were errors creating it? looking at the implementation of `createGitHubClient` it looks like the returned client will be `nil` in that case, which will cause a panic here (right?)

Reply:
hm yeah i am curious what actually causes nil, err b/c even with the erroneous endpoint and fake token it does generate a 'valid' client - but yes, will gate it to prevent panic!

Reply:
looks like you get `nil` [if there was a failed attempt to create an enterprise client](https://github.com/trufflesecurity/trufflehog/blob/62d359eba4c41e76d43cdca832c275795e8c3477/pkg/sources/github/github.go#L647C1-L647C1)